{
    "name": "tensorflow-cpu",
    "cve_id": "CVE-2022-23575",
    "publish_date": "2022-02-04T23:15Z",
    "cwe_id": "CWE-190",
    "owner": "Google Inc.",
    "affected_versions": [
        {
            "version_value": "2.8.0",
            "operator": "<"
        }
    ],
    "references": [
        "https://nvd.nist.gov/vuln/detail/CVE-2022-23575",
        "https://github.com/tensorflow/tensorflow/blob/a1320ec1eac186da1d03f033109191f715b2b130/tensorflow/core/grappler/costs/op_level_cost_estimator.cc#L1552-L1558",
        "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-c94w-c95p-phf8",
        "https://github.com/advisories/GHSA-c94w-c95p-phf8",
        "https://github.com/tensorflow/tensorflow/commit/fcd18ce3101f245b083b30655c27b239dc72221e"
    ],
    "description": "From NVD:\nTensorflow is an Open Source Machine Learning Framework. The implementation of `OpLevelCostEstimator::CalculateTensorSize` is vulnerable to an integer overflow if an attacker can create an operation which would involve a tensor with large enough number of elements. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.\nFrom Github:\n### Impact\nThe [implementation of `OpLevelCostEstimator::CalculateTensorSize`](https://github.com/tensorflow/tensorflow/blob/a1320ec1eac186da1d03f033109191f715b2b130/tensorflow/core/grappler/costs/op_level_cost_estimator.cc#L1552-L1558) is vulnerable to an integer overflow if an attacker can create an operation which would involve a tensor with large enough number of elements:\n```cc\nint64_t OpLevelCostEstimator::CalculateTensorSize(\n    const OpInfo::TensorProperties& tensor, bool* found_unknown_shapes) {\n  int64_t count = CalculateTensorElementCount(tensor, found_unknown_shapes);\n  int size = DataTypeSize(BaseType(tensor.dtype()));\n  VLOG(2) << \"Count: \" << count << \" DataTypeSize: \" << size;\n  return count * size;\n}\n```\nHere, `count` and `size` can be large enough to cause `count * size` to overflow.\n\n### Patches\nWe have patched the issue in GitHub commit [fcd18ce3101f245b083b30655c27b239dc72221e](https://github.com/tensorflow/tensorflow/commit/fcd18ce3101f245b083b30655c27b239dc72221e).\n\nThe fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.",
    "latest_version": "2.8.0",
    "language": "python",
    "license": "Apache-2.0",
    "repository_summary": "TensorFlow is an open source machine learning framework for everyone.",
    "repo_url": "https://github.com/tensorflow/tensorflow",
    "impact": {
        "cvss2_score": 4.0,
        "cvss2_severity": "MEDIUM",
        "vector_string": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
        "av": "NETWORK",
        "ac": "LOW",
        "pr": "LOW",
        "c": "NONE",
        "i": "NONE",
        "a": "HIGH",
        "impact_score": 3.6,
        "exploitability_score": 2.8,
        "cvss3_score": 6.5,
        "cvss3_severity": "MEDIUM",
        "ui": "NONE",
        "s": "UNCHANGED"
    },
    "ochrona_severity_score": 6.5
}